plugins/nf-k8s/src/main/nextflow/k8s/K8sConfig.groovy

```diff
@@ -217,50 +325,27 @@ class K8sConfig implements Map<String,Object> {
      * @return The volume claim name for the given mount path
      */
     String findVolumeClaimByPath(String path) {
-        def result = podOptions.volumeClaims.find { path.startsWith(it.mountPath) }
+        final result = pod.volumeClaims.find { path.startsWith(it.mountPath) }
         return result ? result.claimName : null
     }
 
     @Memoized
     ClientConfig getClient() {
 
-        final result = ( target.client instanceof Map
-                ? clientFromNextflow(target.client as Map, target.namespace as String, target.serviceAccount as String)
-                : clientDiscovery(target.context as String, target.namespace as String, target.serviceAccount as String)
-        )
-
-        if( target.httpConnectTimeout )
-            result.httpConnectTimeout = target.httpConnectTimeout as Duration
+        final result = clientDiscovery(context, namespace, serviceAccount)
```

Comment:
There is a hidden scope `k8s.client` which can be used to define settings for the K8s client directly in Nextflow config. I think the best practice is to define these settings in `.kube/config`. If we want to keep this approach we should explicitly document it.